src/components/Nav.tsx

```diff
@@ -0,0 +1,87 @@
+import { signOut } from "next-auth/react";
+import Link from "next/link";
+import {
+  BarChartIcon,
+  CalendarIcon,
+  DollarSignIcon,
+  HomeIcon,
+  PiggyBankIcon,
+  SettingsIcon,
+  TrendingUpIcon,
+  UserIcon,
+  WalletIcon,
+} from "../lib/svgs";
+
+export default function Nav() {
+  return (
+    <aside className="group fixed right-0 top-1/2 hidden w-20 -translate-y-1/2 transition-all duration-500 ease-in-out hover:w-64 md:flex">
+      <div className="flex-1 overflow-auto py-2">
+        <div className="h-full overflow-hidden">
+          <nav className="flex h-full flex-col items-start px-2 text-base font-medium">
```

Comment:
This array is ideal, one small improvement would be to turn the array into a variable called something like navData, and then map off of that. That would help clean up the JSX in the component and make it more readable.